### ISSUE TITLE
Varying number of results returned when running List tests in parallel

### ISSUE THREAD

**Issue by fwessels** [minio/minio#1191]
While testing for #1186 I ran `mc` a few times for verification purposes. While running `mc` and `listperf` in different terminal windows in parallel to the same minio server (with #1186 patch) I noticed some apparently prematurely terminated responses (and I would guess, the xx001 number of responses is not a coincidence as you are listing per 1000 items...).

```
$ mc ls myminio/frank/ | wc -l
  102983
$ mc ls myminio/frank/ | wc -l
   92001
$ mc ls myminio/frank/ | wc -l
    2001
 $ mc ls myminio/frank/ | wc -l
  102983
```

**Comment by harshavardhana** [comment]
> $ mc ls myminio/frank/ | wc -l
>   102983
> $ mc ls myminio/frank/ | wc -l
>    92001
> $ mc ls myminio/frank/ | wc -l
>     2001
>  $ mc ls myminio/frank/ | wc -l
>   102983

Yes this is fixed, i have seen this issue as well :-)

**Comment by fwessels** [comment]
Ok, great.

**Comment by harshavardhana** [comment]
> Ok, great.

Spoke too soon, the patch looks like still has this issue - pinged @balamurugana

**Comment by fwessels** [comment]
I observed the issue with the patch applied, not sure if it also occurs without the patch.

**Comment by harshavardhana** [comment]
Closing this, since is fixed in new patchset at #1186
